compaction/task_manager_module.cc

```diff
@@ -453,14 +453,17 @@ future<> table_cleanup_keyspace_compaction_task_impl::run() {
 }
 
 future<> offstrategy_keyspace_compaction_task_impl::run() {
-    _needed = co_await _db.map_reduce0([&] (replica::database& db) -> future<bool> {
+    bool res = co_await _db.map_reduce0([&] (replica::database& db) -> future<bool> {
         bool needed = false;
         tasks::task_info parent_info{_status.id, _status.shard};
         auto& module = db.get_compaction_manager().get_task_manager_module();
         auto task = co_await module.make_and_start_task<shard_offstrategy_keyspace_compaction_task_impl>(parent_info, _status.keyspace, _status.id, db, _table_infos, needed);
         co_await task->done();
         co_return needed;
     }, false, std::plus<bool>());
+    if (_needed) {
+        *_needed = res;
```

Comment:
Is the "needed" an output parameter? If so can we make it really an output? return `future<std::tuple<task, bool>>`, or better a struct with two fields so we can understand what it means.

Reply:
Yes, that's an output parameter. Currently `task_manager::task`s don't support returning value from their `run` method. That's because `run()` isn't called directly, but rather in some background fiber. Its result can be awaited with shared future.

It could probably be changed, but that would significantly complicate the code and, at least for now, output parameters are rarely needed. I'm not sure that's worth it. Anyway, that's out of scope for this PR.